preview-maker.js

```diff
@@ -1,184 +1,392 @@
 const NEW_TEMPLATE = Symbol();
 
 const templates = {
-    [REGULAR_TEMPLATE]: fs.readFileSync(`./theme-preview.svg`, `utf8`),
-    [NEW_TEMPLATE]: fs.readFileSync(`./new-preview.svg`, `utf8`),
+  [MINIMALISTIC_TEMPLATE]: fs.readFileSync(`./new-theme-preview.svg`, `utf8`),
+  [REGULAR_TEMPLATE]: fs.readFileSync(`./theme-preview.svg`, `utf8`),
+  [NEW_TEMPLATE]: fs.readFileSync(`./new-preview.svg`, `utf8`),
 };
 
 const WALLPAPERS_AMOUNT = 32;
 const wallpapers = [];
 
 for (let index = 0; index < WALLPAPERS_AMOUNT; index++) {
-    const wallpaper = fs.readFileSync(
-        `./wallpapers/${index}.jpg`,
-        `binary`,
-    );
+  const wallpaper = fs.readFileSync(`./wallpapers/${index}.jpg`, `binary`);
 
-    wallpapers.push(wallpaper);
+  wallpapers.push(wallpaper);
 }
 
-const get = (node, className, tag) => Array.from(node.getElementsByTagName(tag))
-    .filter((element) => (
-        element.getAttribute
-        && element.getAttribute(`class`) === className
-    ));
+const get = (node, className, tag) =>
+  Array.from(node.getElementsByTagName(tag)).filter(
+    (element) =>
+      element.getAttribute && element.getAttribute(`class`) === className
+  );
 
 const getElementsByClassName = (node, className) => [
-    ...get(node, className, `rect`),
-    ...get(node, className, `circle`),
-    ...get(node, className, `path`),
-    ...get(node, className, `g`),
-    ...get(node, className, `polygon`),
-    ...get(node, className, `image`),
-    ...get(node, className, `tspan`),
-    ...get(node, className, `stop`),
+  ...get(node, className, `rect`),
+  ...get(node, className, `circle`),
+  ...get(node, className, `path`),
+  ...get(node, className, `g`),
+  ...get(node, className, `polygon`),
+  ...get(node, className, `image`),
+  ...get(node, className, `tspan`),
+  ...get(node, className, `stop`),
 ];
+const calculateAccentColor = (colors) => {
+  let colorsQuantity = new Map();
+  let max = 0;
+  let accentHue;
+
+  for (const color of colors) {
+    if (color != 0) {
+      colorsQuantity.set(color, (colorsQuantity.get(color) ?? 0) + 1);
+
+      if (colorsQuantity.get(color) > max) {
+        accentHue = color;
+        max = colorsQuantity.get(color);
+      }
+    }
+  }
+  const colorsHue = colors.filter((color) => colorsQuantity.get(color) == max);
+  colorsHue.sort((a, b) => a - b);
+
+  if (colorsHue.length > 1) {
+    let minDifference = 360;
+    for (let i = 0; i < colorsHue.length - 1; i++) {
+      const outerHue = colorsHue[i];
+      const innerHue = colorsHue[i + 1];
+      if (Math.abs(outerHue - innerHue) < minDifference) {
+        minDifference = Math.abs(outerHue - innerHue);
+        accentHue = (outerHue + innerHue) / 2;
+      }
+    }
+  }
 
-const fill = (node, color) => {
-    const { red, green, blue, alpha } = color;
-    const rgba = `rgba(${red}, ${green}, ${blue}, ${alpha / 255})`;
-
+  return accentHue;
+}
+const getDifference = (chooseHsl, windowBackgroundWhite, argument) => {
```

Comment:
`argument`? what? the code is wrong for saturation and lightness, and there's no need to try to re-use this function because for saturation and lightness, you can just do `Math.abs(chooseHsl.saturation - windowBackgroundWhite.saturation)`.